internal/action/bufpane.go

```diff
@@ -151,24 +151,22 @@ func BufMapEvent(k Event, action string) {
 	}
 	bufAction := func(h *BufPane, te *tcell.EventMouse) bool {
 		cursors := h.Buf.GetCursors()
-		success := true
 		for i, a := range actionfns {
-			innerSuccess := true
+			success := true
 			for j, c := range cursors {
 				if c == nil {
 					continue
 				}
 				h.Buf.SetCurCursor(c.Num)
 				h.Cursor = c
 				if i == 0 || (success && types[i-1] == '&') || (!success && types[i-1] == '|') || (types[i-1] == ',') {
-					innerSuccess = innerSuccess && h.execAction(a, names[i], j, te)
+					success = success && h.execAction(a, names[i], j, te)
```

Comment:
I don't think this is equivalent to the previous logic. Also this seems buggy.
1. In the previous logic, if the action failed for the given cursor, we still keep executing it for the remaining cursors, before stopping the action chain. In your new logic, we stop as soon as the action failed, so some cursors may remain unserved.
2. More importantly: you reset `success` to true at each iteration of the outer loop, i.e. for each next action in the chain. So, `&` and `|` are not gonna work correctly at all?

Moreover: just removing this `success := true` from line 155 wouldn't be enough to fix that: it would mean that once `success` became false, it never becomes true again, so e.g. with `foo|bar&baz` chain, if `foo` but `bar` succeeded, `baz` is not executed: it still thinks that the previous action failed.

BTW, FYI `innerSuccess` was introduced in https://github.com/zyedidia/micro/commit/037c3c993fe81a258ed273fccfdfcb9183d7fafb as a part of the fix for #1721. Though I haven't analyzed what was the issue #1721 about and how exactly it relates to `innerSuccess`. But it looks like `innerSuccess` is needed for correct logic anyway?

Reply:
Yep, logic and behavior changed.
I removed the commit and leave it as it is so far. Especially for now I will make a brake...it's too late for that and will become even worse. :wink: